test/functional/http/get/blocks.js

```diff
@@ -4,405 +4,402 @@ var node = require('../../../node.js');
+				})
+				.then(function () {
+					return getJsonForKeyPromise(url + params.join('&'));
+				})
+				.then(function (result) {
+					node.expect(result).to.eql(null);
+				});
 		});
 	});
 
-	describe('/getNethash', function () {
+	describe('/', function () {
 
-		it('should be ok', function (done) {
-			http.get('/api/blocks/getNethash', function (err, res) {
-				node.expect(res.body).to.have.property('success').to.be.ok;
-				node.expect(res.body).to.have.property('nethash').to.be.a('string');
-				node.expect(res.body.nethash).to.equal(node.config.nethash);
-				done();
-			});
-		});
-	});
+		describe('id', function () {
 
-	describe('/getMilestone', function () {
+			it('using genesisblock id should return the result', function () {
+				var id = '6524861224470851795';
+				var params = [
+					'id=' + id
+				];
 
-		it('should be ok', function (done) {
-			http.get('/api/blocks/getMilestone', function (err, res) {
-				node.expect(res.body).to.have.property('milestone').to.be.a('number');
-				done();
+				return getBlocksPromise(params).then(function (res){
+					node.expect(res).to.have.property('blocks').to.be.an('array').that.have.nested.property('0.id').equal(id);
+				});
 			});
-		});
-	});
 
-	describe('/getReward', function () {
+			it('using unknown id should return empty blocks array', function () {
+				var id = '9928719876370886655';
+				var params = [
+					'id=' + id
+				];
 
-		it('should be ok', function (done) {
-			http.get('/api/blocks/getReward', function (err, res) {
-				node.expect(res.body).to.have.property('reward').to.be.a('number');
-				done();
+				return getBlocksPromise(params).then(function (res){
+					node.expect(res).to.have.property('blocks').to.be.an('array').and.to.be.empty;
+				});
 			});
 		});
-	});
-
-	describe('/getSupply', function () {
 
-		it('should be ok', function (done) {
-			http.get('/api/blocks/getSupply', function (err, res) {
-				node.expect(res.body).to.have.property('supply').to.be.a('number');
-				done();
+		describe('height', function () {
+
+			it('using correct params should be ok', function () {
+				var params = [
+					'height=' + block.blockHeight
+				];
+
+				return getBlocksPromise(params).then(function (res){
+					node.expect(res).to.have.property('blocks').that.is.an('array');
+					node.expect(res.blocks.length).to.equal(1);
+					node.expect(res.blocks[0]).to.have.property('id');
+					node.expect(res.blocks[0]).to.have.property('previousBlock');
+					node.expect(res.blocks[0]).to.have.property('totalAmount');
+					node.expect(res.blocks[0]).to.have.property('totalFee');
+					node.expect(res.blocks[0]).to.have.property('generatorId');
+					node.expect(res.blocks[0]).to.have.property('confirmations');
+					node.expect(res.blocks[0]).to.have.property('blockSignature');
+					node.expect(res.blocks[0]).to.have.property('numberOfTransactions');
+					node.expect(res.blocks[0].height).to.equal(block.blockHeight);
+					block.id = res.blocks[0].id;
```

Comment:
Why are assigning blocks returned in response to block variable?

Reply:
I don't have an answer for that, the code seems to be new but it is not - @diego-G changed the indentation